FoodBookApp/ContentView.swift

```diff
@@ -67,12 +68,35 @@ struct ContentView: View {
                     .tabItem { Label("Bookmarks", systemImage: "book") }
                     .tag(Tabs.bookmarks)
                 
-            }.navigationTitle(selectedTab.formattedTitle)
+            }
+            .navigationTitle(selectedTab.formattedTitle)
+            .modifier(SearchableModifier(isSearchable: selectedTab == .browse, text: $searchText))
             
         }
     }
 }
 
+struct SearchableModifier: ViewModifier { // FIXME: Refactor, where should I have this modifier?
+    let isSearchable: Bool
+    @Binding var text: String
+
+    func body(content: Content) -> some View {
+        if isSearchable {
+            return content
+                .searchable(text: $text)
+                .eraseToAnyView()
+        } else {
+            return content.eraseToAnyView()
+        }
+    }
+}
+
+extension View {
```

Comment:
just a curiosity, what does this does?

Reply:
Great question! This is helping the ContentView not show the searchable bar for every view (ForYou, Bookmarks) but just the Browse view. I have the comment of refactoring because I am not sure this should be a part of this file but its currently here for testing.